### PR TITLE
Added docker build option

### PR DESCRIPTION
**Pull request by drallgood**
I found it easier to run mattermost integrations as docker containers (allows you to add them to docker-compose as well)
Thus, I enabled this project to be run in a container.

A few minor changes were necessary:
- Introduced a `conf`directory that can be mounted externally
- Moved config.json and mattermost.json to this directory
- Also added a json template file for convenience

Please let me know whether there are any changes to this PR that I should make (not as familiar with go)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -540,15 +540,15 @@ func (b *Bot) Logf(msg string, args ...interface{}) {
...
+		DataPath:      "conf/mattermost.json",
 		Channel:       "town-square",
 		DebugChannel:  "test",
 		MaxTweets:     20,
 		CheckInterval: 120,
 	}
 
 	// Parse cmdline flags
-	flag.StringVar(&confPath, "config", "config.json",
+	flag.StringVar(&confPath, "conf/config", "conf/config.json",
```
Comment (bwesterb) [inline main.go]: This is a backwards incompatible change.  I don't want people's current installation to break when they do an update.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,3 @@
 /twittermost
-/twittermost.json
-/config.json
+*/mattermost.json
```
Comment (bwesterb) [inline .gitignore:2]: Ah, yeah, so I wrote `mattermost.json` instead of `twittermost.json`.  ^^'

**Comment by bwesterb** [comment]
I'm not a docker adapt myself, is it possible to move all the docker files into a subdirectory so as not to clutter the root folder too much?

**Comment by drallgood** [comment]
@bwesterb 
Not sure which files you mean, since there's only the Dockerfile
It would be possible to move it but I wouldn't recommend it since this makes things less obvious

**Comment by bwesterb** [comment]
and the `.dockerignore`, but you're right... it's only two files.

**Review by bwesterb — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (drallgood) [reply .gitignore:2]: yep

File: main.go (hunk 1 above)
Comment (drallgood) [reply main.go]: hmm.. yeah.. that's what I didn't like about this. I'm not too well versed in Go. Is there a way that we can keep this backwards compatible but still have the config in a folder?

Docker only allows you to mount folders and the config would need to be customized. Hence my moving it into a folder.
This way we can even publish a pre-built docker image to the docker registry that people can use.

**Comment by drallgood** [comment]
The `.dockerignore` has to be in the root (just like `.gitignore`). It's not strictly necessary but prevents files/folders from being added to the image. In this case I excluded the conf folder since we don't want to bake the config into the image.

File: main.go (hunk 1 above)
Comment (bwesterb) [reply main.go]: I barely know anything about docker, but couldn't you just specify the configuration file path in the `Dockerfile`?

Comment (drallgood) [reply main.go]: that should be possible, yes. 
Give me a few minutes to give this a try

**Comment by drallgood** [comment]
Alright. I changed the Dockerfile. 
Only drawback is, that now the default for non-docker installs is to still have the config file in the root folder instead of the `conf` folder (otherwise you would need to specify the `-config`option every time)

**Comment by bwesterb** [comment]
I would prefer if you would rename `config.template` to `config.json.example` and move it into the root.  (Then just add instructions in the README to create a conf directory.)  If you want I can do this as well for you.

**Comment by drallgood** [comment]
I can do it... However, what happens to the conf directory?
It has to be present for two reasons:
1) I don't want people to create it every time. Too much hassle and too many things that can go wrong
2) the mattermost.json is placed there if you use the provided template (I have to do this so that it is persisted in docker environments).

Can't we just declare the conf directory the new default location for the config(s) and just make sure all legacy installations still work???

**Comment by bwesterb** [comment]
Ok, as long as legacy installations still work (without changes). :)

**Comment by drallgood** [comment]
How about this :)

**Comment by bwesterb** [comment]
Ugly!  But I guess it's the best we can do without ruining someone's day.

**Pull request merged**